lib/src/Base/Graph/GraphImplementation.cxx

```diff
@@ -175,6 +179,12 @@ GraphImplementation::DrawableCollection GraphImplementation::getDrawables() cons
 void GraphImplementation::setDrawables(const DrawableCollection & drawableCollection)
 {
   drawablesCollection_ = drawableCollection;
+  for (UnsignedInteger i = 0; i < drawableCollection.getSize(); ++i)
+  {
+    Drawable drawable = drawablesCollection_[i];
+    if (!drawable.isColorExplicitlySet())
+      drawable.setColor(DrawableImplementation::BuildDefaultPalette(i + 1)[i]);
```

Comment:
the palette is build several times in the loop, build it once with fixed size

Reply:
For future continuous palettes, the result will be different depending on whether we add drawable by drawable or a set of drawables.

Reply:
that was exactly my point in the previous remark, but it should not, here we still apply the last color of the palette so the full range of colors will never be explored

Reply:
but it looks like the current 2 palettes do not actually depend on the size, so the current version is ok just inefficient, maybe we want to use a new GetPaletteAtIndex(i) which returns the i-th color

Reply:
For continuous palettes, the ith color only makes sense if we know the total number of colors to assign because the colors will be distributed uniformly over the entire scale.
For a GrayScale palette BuildPalette(i)[i-1] will systematically give black while BuildPalette(N)[i-1] will give a gray depending on N.

Reply:
> the palette is built several times in the loop, build it once with fixed size

Done in commit https://github.com/openturns/openturns/pull/2422/commits/34c5888581e55a4136075cff7cfd04b1bbd725cb